doc/source/asyncio/guide/pattern.rst

```diff
@@ -0,0 +1,252 @@
+            r = requests.get(url)
+            return r.json()
+
+The end user could be eager to execute this HTTP call in an asynchronous way
+to avoid blocking his program on a network call. The end user may use your
+lib to monitor numerous instances and so he would want to avoid loosing time.
+
+The end user may use Eventlet, to monkey patch all is stack at runtime.
+Then the request call would become async::
+
+    import eventlet
+    eventlet.monkey_patch()
+    from mystack import Client # non-blocking now
+    client = Client()
+    client.request("https://rest.api.xyz/vm/12345")
+
+But ``mystack`` could be also used the same way on services that do not
+rely on Eventlet, where blocking call are not a problem, or where the
+service manage async things internally by using threading or process,
+who know...
+
+The ``mystack`` lib is currently agnostic to network programming model used
+by its end users - the environment defined by the service importing
+``mystack``.
+
+The Pattern
+-----------
+
+**The proposed pattern may be used either for migrating libraries or standard
+applications.** The semantic will remains the same in both cases.
+
+.. warning::
+
+    Using the following pattern require to use the :mod:`Asyncio Hub
+    <eventlet.hubs.asyncio>`
+
+    :ref:`understanding_hubs`
+
+Here is a refactor of the previous ``mystack`` code which is now based on the
+AsyncIO facade pattern::
+
+    # mystack.client
+    import asyncio
+    import aiohttp
+
+
+    class AsyncClient():
+        headers = {
+            'User-Agent': 'async_client',
+        }
+
+        async def request(self, url):
+            async with aiohttp.ClientSession(headers=self.headers) as session:
+                async with session.get(url) as response:
+                    res = await response.json()
+                    return res
+
+
+    class Client(AsyncClient):
+        def __init__(self):
+            self.headers = {
+                'User-Agent': 'sync_client',
+            }
+
+        def _iter_coroutine(self, coro):
+            loop = None
+            try:
+                loop = asyncio.get_running_loop()
+                except RuntimeError:
+                    loop = asyncio.new_event_loop()
+                    return loop.run_until_complete(coro)
+
+        def request(self, url):
+            return self._iter_coroutine(super().request(url))
+
+This module now allow to use either a synchronous or an asynchronous client.
+Behind the scene the synchronous client will call the asynchronous client.
+
+You can observe that we switched the `User-Agent` header depending on the
+client used by the end users. It will allow you to observe the difference if
+you decide to run this example.
+
+If people want to use the asynchronous client, then their code will be
+dotted with `async` and `await` keywords, making their asynchronous calls
+explicit, not like with Eventlet.
+
+If people want to use the synchronous client, then their code will remains
```

Comment:
Yes, but, this is _different_ than using Eventlet, because it blocks, which means you need to use regular threads. The blocking version is no longer usable in Eventlet.